sklearn/linear_model/coordinate_descent.py

```diff
@@ -729,6 +737,9 @@ def fit(self, X, y, check_input=True):
         self.coef_, self.dual_gap_ = map(np.squeeze, [coef_, dual_gaps_])
         self._set_intercept(X_offset, y_offset, X_scale)
 
+        # workaround since _set_intercept will cast self.coef_ into float64
+        self.coef_ = np.asarray(self.coef_, dtype=X.dtype)
```

Comment:
I originally thought the input data is explicitly cast to float64 before the call to `_preprocess_data` for the other models, but then found that some are not.

I am afraid of some models assume float32 data will be transformed into float64 after `_preprocess_data`.